docs/configuring-a-resource.md

```diff
@@ -0,0 +1,410 @@
+some effort, however, with the latest Crossplane code generation tooling, it is
+now possible to [generate reference resolution methods] by just adding some
+marker on the fields. Now, the only manual step for generating cross resource
+references is to provide which field of a resource depends on which information
+(e.g. `id`, `name`, `arn` etc.) from the other.
+
+In Terrajet, we have a [configuration] to provide this information for a field:
+
+```go
+// Reference represents the Crossplane options used to generate
+// reference resolvers for fields
+type Reference struct {
+    // Type is the type name of the CRD if it is in the same package or
+    // <package-path>.<type-name> if it is in a different package.
+    Type string
+    // Extractor is the function to be used to extract value from the
+    // referenced type. Defaults to getting external name.
+    // Optional
+    Extractor string
+    // RefFieldName is the field name for the Reference field. Defaults to
+    // <field-name>Ref or <field-name>Refs.
+    // Optional
+    RefFieldName string
+    // SelectorFieldName is the field name for the Selector field. Defaults to
+    // <field-name>Selector.
+    // Optional
+    SelectorFieldName string
+}
+```
+
+For a resource that we want to generate, we need to check its argument list in
+Terraform documentation and figure out which field needs reference to which
+resource.
+
+Let's check [iam_access_key] as an example. In the argument list, we see the
+[user] field which requires a reference to a IAM user. So, we need to the
+following referencing configuration:
+
+```go
+func Configure(p *config.Provider) {
+    p.AddResourceConfigurator("aws_iam_access_key", func (r *config.Resource) {
+        r.References["user"] = config.Reference{
+            Type: "User",
+        }
+    })
+}
+```
+
+Please note the value of `Type` field needs to be a string representing the Go
+type of the resource. Since, `AccessKey` and `User` resources are under the same
+go package, we don't need to provide the package path. However, this is not
+always the case and referenced resources might be in different package. In that
+case, we would need to provide the full path. Referencing to a [kms key] from
+`aws_ebs_volume` resource is a good example here:
+
+```go
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("aws_ebs_volume", func(r *config.Resource) {
+		r.References["kms_key_id"] = config.Reference{
+			Type: "github.com/crossplane-contrib/provider-tf-aws/apis/kms/v1alpha1.Key",
+		}
+	})
+}
+```
+
+### Additional Sensitive Fields and Custom Connection Details
+
+Crossplane stores sensitive information of a managed resource in a Kubernetes
+secret, together with some additional fields that would help consumption of the
+resource, a.k.a. [connection details].
+
+In Terrajet, we already [handle sensitive fields] that are marked as sensitive
+in Terraform schema and no further action required for them. However, we still
```

Comment:
I think we should tell them that if the field is marked as sensitive it's automatically removed from CRD and published only in the connection details secret.